### PR TITLE
[GR-1209] Fixes to bcl2barcode index calling

### PR DESCRIPTION
**Pull request by slazicoicr**

### CODE REVIEW

File: application/dash_application/views/bcl2barcode.py [hunk 1]
```diff
@@ -61,74 +136,100 @@ def trim(row, index_name, len_name):
...
+bcl2barcode['Index2MatchedCount'] = bcl2barcode['Index2Matched'].apply(len)
+bcl2barcode['Index1MatchedSequence'] = bcl2barcode['Index1Matched'].apply(lambda x: x[0] if len(x) else '')
+bcl2barcode['Index2MatchedSequence'] = bcl2barcode['Index2Matched'].apply(lambda x: x[0] if len(x) else '')
+
+bcl2barcode["Classify"] = bcl2barcode.apply(classify, axis=1)
+
+# Sum all bcl2barcode counts that come from the same Pinery barcode
+all_known = bcl2barcode[(bcl2barcode["Classify"] == "Known")]
+all_known = all_known.groupby(
+    [PINERY_COL.SequencerRunName, PINERY_COL.LaneNumber, 'Index1MatchedSequence', 'Index2MatchedSequence', 'IndexStrategy']
```
Comment (apmasell) [inline application/dash_application/views/bcl2barcode.py:207]: This is real barcode from Pinery (i.e., after 10X expansion), right?

File: application/dash_application/views/bcl2barcode.py [hunk 2]
```diff
@@ -61,74 +136,100 @@ def trim(row, index_name, len_name):
...
+    known_data_table,
+    known_data_table_dual[~known_data_table_dual['studyTitle'].isna()]
+])
+unknown_data_table = pandas.concat([
+    unknown_data_table,
+    known_data_table_dual[known_data_table_dual[PINERY_COL.StudyTitle].isna()],
+], join="inner")
 
-dual_unknown = dual_index[dual_index['studyTitle'].isna()][bcl2barcode.columns]
+known_data_table.to_csv('del.csv')
```
Comment (apmasell) [inline application/dash_application/views/bcl2barcode.py]: Testing?

**Review by apmasell — APPROVED** [review]
That reads like pea soup, but I think that's unavoidable.

File: application/dash_application/views/bcl2barcode.py (hunk 2 above)
Comment (slazicoicr) [reply application/dash_application/views/bcl2barcode.py]: Yep

File: application/dash_application/views/bcl2barcode.py (hunk 1 above)
Comment (slazicoicr) [reply application/dash_application/views/bcl2barcode.py:207]: Correct. 10X barcode correction is one of the first steps.

**Review by avarsava — APPROVED** [review]
Great comments on what you're doing. More explanation of _why_ would be appreciated for if we need to touch this a long time from now - not sure I understand what a 'collision index' is

**Review by mlaszloffy — APPROVED** [review]

**Comment by slazicoicr** [comment]
@avarsava I expanded the comments. Let me know if there is anything else that is confusing.

**Comment by avarsava** [comment]
> @avarsava I expanded the comments. Let me know if there is anything else that is confusing.

Looks great, thanks :+1:

**Pull request merged**